nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -974,6 +993,9 @@ private ValidationResult validateControllerServiceApi(final PropertyDescriptor d
         return null;
     }
 
+    protected void analyze() {
```

Comment:
We should avoid method names that are ambiguous. It's very unclear here what sort of analysis should be performed.

Reply:
That's true but I'm struggling to come up with a good name that's not too long.

This type of analysis conceptually called `FlowAnalysis`. But something like `perfromFlowAnalysis` would suggest that this component wants to do a full-scale analysis. `performFlowAnalysisOnThisComponent` feels too long.
Maybe `performFlowAnalysisOnThis`?